src/main/java/com/mathworks/ci/MatlabReleaseInfo.java

```diff
@@ -107,7 +114,22 @@ public boolean verLessThan(double version) throws MatlabVersionNotFoundException
                 else if(!this.matlabRoot.exists()){
                     throw new NotDirectoryException("Invalid matlabroot path");
                 }else {
-                    versionInfoCache.putAll(VERSION_OLDER_THAN_17A);
+                	FilePath contentFile = new FilePath(this.matlabRoot,"toolbox/matlab/general/Contents.m");
+                	InputStream in = contentFile.read();
+                	BufferedReader br = new BufferedReader(new InputStreamReader(in));
+                	
+                	//Skip first line and capture the second line.
+                	br.readLine();
+                	String versionLine = br.readLine();
+                	
+                	String actualVersion = VERSION_16B;
```

Comment:
Id say its better to use a Null here rather than the 16b default. This lets us know that there was a problem as opposed to giving the wrong version

Reply:
Done.